packages/block-editor/src/store/reducer.js

```diff
@@ -983,8 +983,35 @@ export const blockListSettings = ( state = {}, action ) => {
 	return state;
 };
 
+export function lastBlockAttributesChanges( state, action ) {
+	switch ( action.type ) {
+		case 'UPDATE_BLOCK':
+			if ( ! action.updates.attributes ) {
+				return null;
+			}
+			return {
+				[ action.clientId ]: action.updates.attributes,
+			};
+
+		case 'UPDATE_BLOCK_ATTRIBUTES':
+			return {
+				[ action.clientId ]: action.attributes,
+			};
+
+		case 'RESET_BLOCKS':
+		case 'INSERT_BLOCKS':
+		case 'RECEIVE_BLOCKS':
+		case 'REPLACE_BLOCKS':
+			return getFlattenedBlockAttributes( action.blocks );
+	}
+
+	return null;
```

Comment:
yes, I didn't give it too much thoughts but yeah the idea is that we don't care about the value after. I don't think it would harm if we return "state" but I wonder if we'd need to "reset" the value in some actions I might have missed.

Reply:
Tbh, this "last changes" thing was consider very experimental. If we follow the approach of this PR, I wouldn't mind if we do a full sync on update. Maybe the performance impact is small? we should measure.